### PR TITLE
#54 - Added start upload endpoint

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #54 
Added start upload endpoint

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#80](https://github.com/artipie/docker-adapter/issues/80) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/b1aa0d35-44b9-476a-9c91-1bff44c012e6); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/docker/http/StartUploadTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            )
+        );
+    }
+
+    /**
+     * Header matcher.
+     *
+     * @since 0.2
+     */
+    private static class IsHeader extends TypeSafeMatcher<Map.Entry<String, String>> {
```
Comment (Vatavuk) [inline src/test/java/com/artipie/docker/http/StartUploadTest.java:107]: @olegmoz I think it would be good to make this class public so we can reuse it.

Comment (olegmoz) [reply src/test/java/com/artipie/docker/http/StartUploadTest.java:107]: @Vatavuk good idea, done it. Probably will add it to HTTP module later

**Comment by 0crat** [comment]
Job was finished in 45 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**